server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -276,17 +292,27 @@ public Map<String, Long> buildQueryBreakdownMap(
                 );
 
                 // query start/end time for a TimingType is min/max of start/end time across slices for that TimingType
-                queryTimingTypeEndTime = Math.max(queryTimingTypeEndTime, sliceBreakdown.getValue().get(sliceEndTimeForTimingType));
-                queryTimingTypeStartTime = Math.min(queryTimingTypeStartTime, sliceBreakdown.getValue().get(sliceStartTimeForTimingType));
+                queryTimingTypeEndTime = Math.max(
+                    queryTimingTypeEndTime,
+                    sliceBreakdown.getValue().getOrDefault(sliceEndTimeForTimingType, Long.MIN_VALUE)
+                );
+                queryTimingTypeStartTime = Math.min(
+                    queryTimingTypeStartTime,
+                    sliceBreakdown.getValue().getOrDefault(sliceStartTimeForTimingType, Long.MAX_VALUE)
+                );
                 queryTimingTypeCount += sliceBreakdownTypeCount;
             }
+            assert queryTimingTypeStartTime != Long.MAX_VALUE && queryTimingTypeEndTime != Long.MIN_VALUE : "Unexpected queryTimingType "
```

Comment:
This is an assert which is verifying that both variables are set to non default values as we expect this function to be called only when there is valid leaf contexts for slices.

Reply:
@sohami  assertions are great but they could be turned off, so for calculation you should better check the conditions nonetheless since no exception will be raised but bogus results returned.

Reply:
Have added the check here and above to throw `exception` instead to return 0 in these unexpected use cases. Returning `0` will be same as returning bogus values and can hide these cases